lib/sycamore/sycamore/transforms/aryn_partitioner.py

```diff
@@ -0,0 +1,50 @@
+from typing import BinaryIO
+from collections.abc import Mapping
+import requests
+import json
+from sycamore.data.element import create_element
+
+_DEFAULT_ARYN_PARTITIONER_ADDRESS = "https://api.aryn.cloud:8000/v1/partition"
+
+
+class ArynPDFPartitionerException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class ArynPDFPartitioner:
+    @staticmethod
+    def partition_pdf(
+        file: BinaryIO,
+        token: str,
+        aryn_partitioner_address=_DEFAULT_ARYN_PARTITIONER_ADDRESS,
+        threshold: float = 0.4,
+        use_ocr: bool = False,
+        ocr_images: bool = False,
+        ocr_tables: bool = False,
+        extract_table_structure: bool = False,
+        extract_images: bool = False,
+    ):
+        options = {
+            "threshold": threshold,
+            "use_ocr": use_ocr,
+            "ocr_images": ocr_images,
+            "ocr_tables": ocr_tables,
+            "extract_table_structure": extract_table_structure,
+            "extract_images": extract_images,
+        }
+        files: Mapping = {"pdf": file, "options": json.dumps(options).encode("utf-8")}
```

Comment:
why are you sending Mapping as a file? Can we send options in the body instead?

Reply:
mypy expects the `files` variable to be a `Mapping` not a `dict` when used in `requests.post(...)`.  Let's keep the API as-is: we need to upload both `options` and the pdf itself.

Reply:
Sounds like there's some confusion here. A `dict` is a `Mapping`. If a function takes a `Mapping`, you can pass it anything that has `__getitem__`, `__iter__` and `__len__` (of the proper type, if it specifies type variables, like `Mapping[str, str]`)—including a `dict`.

Think of `Mapping` as an interface of objects that quack like a `dict`. (There's also `MutableMapping` which a function should take if it doesn't absolutely require it to be a `dict`, but it wants to modify it.)

Reply:
I agree a `dict` is a `Mapping`. For some reason mypy does not see the `dict[str, BinaryIO | bytes]` as a `Mapping[str, SupportsRead[bytes]]`, so I annotated it which silences the error.

`lib/sycamore/sycamore/transforms/aryn_partitioner.py:38: error: Argument "files" to "post" has incompatible type "dict[str, Iterable[object]]"; expected "Mapping[str, SupportsRead[str | bytes] | str | bytes | tuple[str | None, SupportsRead[str | bytes] | str | bytes] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, Mapping[str, str]]] | Iterable[tuple[str, SupportsRead[str | bytes] | str | bytes | tuple[str | None, SupportsRead[str | bytes] | str | bytes] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str] | tuple[str | None, SupportsRead[str | bytes] | str | bytes, str, Mapping[str, str]]]] | None"  [arg-type]`

Reply:
I think the issue is that `bytes` is not a `SupportsRead[bytes]`—I don't think `bytes` has a `read` method. So in this case I think mypy is correct. It passes as a `Mapping` only because that is (an unsafe, --strict would complain) alias for `Mapping[Any,Any]`. Does the code actually work if passed bytes? And the annotation of the called function is not unnecessarily strict?